### PR TITLE
Related records

### PR DESCRIPTION
**Pull request by michamos**
* INCOMPATIBLE Adds `related_records` for hep, experiments, journals and
institutions instead of the previous special cases (`succeeding_entry`,
`related_experiments`, `relation`, `related_institutes` respectively)
and the corresponding `related_record` element.

This is what we discussed with @annetteholtkamp and @jacquerie.

### CODE REVIEW

File: inspire_schemas/records/elements/related_record.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+$schema: http://json-schema.org/schema#
+additionalProperties: false
+dependencies:
+    record:
+        oneOf:
+        -   required:
+            - relation
+        -   required:
+            - other_relation
```
Comment (kaplun) [inline inspire_schemas/records/elements/related_record.yml]: Woah! Havey you checked with @harunurhan? Will it break current editor?

File: inspire_schemas/records/elements/related_record.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+description: |-
+    :MARC: ``78002``, ``78502``, ``78708```and ``510`` (depending on type of
+        record and :ref:`related_record.json#/properties/relation`)
+properties:
+    curated_relation:
+        type: boolean
+    other_relation:
```
Comment (kaplun) [inline inspire_schemas/records/elements/related_record.yml]: Reading `other_relation` I am not understanding this has to be used if `relation` does not fit.

What about `other_relation_type` or `relation_other`?

File: tests/integration/fixtures/experiments_example.json [hunk 3]
```diff
@@ -1,122 +1,139 @@
...
     "core": true,
-    "date_approved": "dddd-dd-dd",
-    "date_cancelled": "dddd-dd-dd",
-    "date_completed": "dddd-dd-dd",
-    "date_proposed": "dddd-dd-dd",
-    "date_started": "dddd-dd-dd",
-    "deleted": true,
+    "date_approved": "9433-07-69",
+    "date_cancelled": "0653-71-44",
+    "date_completed": "5802-68-29",
```
Comment (kaplun) [inline tests/integration/fixtures/experiments_example.json]: These dates are bit ugly... would be great to have real valid ones... Can be fixed in a different issue.

File: inspire_schemas/records/elements/related_record.yml (hunk 1 above)
Comment (michamos) [inline inspire_schemas/records/elements/related_record.yml]: it was done this way on purpose not to break the editor.
Morally speaking, we want to have a big
```yaml
allOf:
    - oneOf:
        - properties:
            relation:
                # definition of relation
        - properties:
            other_relation:
                # definition of other_relation
    - properties:
        record:
            # definition of record
```
but **that** breaks the record editor.
Instead, I used a hack by putting dependencies on `record` which is a required field. To be valid, `record` should be present, which in turns triggers its dependencies. So the effect is the same, but the editor does not have to deal with it before validation.

File: tests/integration/fixtures/experiments_example.json (hunk 3 above)
Comment (michamos) [inline tests/integration/fixtures/experiments_example.json]: before it was producing `dddd-dd-dd` always, so this is already better I would say.

File: inspire_schemas/records/elements/related_record.yml (hunk 2 above)
Comment (michamos) [inline inspire_schemas/records/elements/related_record.yml]: It is not clear to me whether inverting the two words or adding `type` caries a much clearer meaning, but if you think so, I can change it. Maybe `relation_type`/`other_relation_type` for consistency then?

**Comment by michamos** [comment]
@kaplun I addressed your comments.

**Pull request merged**